samza-core/src/main/java/org/apache/samza/job/model/ContainerModel.java

```diff
@@ -87,6 +103,6 @@ public boolean equals(Object obj) {
   }
 
   public int compareTo(ContainerModel other) {
-    return containerId - other.getContainerId();
+    return processorId.compareTo(other.getProcessorId());
```

Comment:
Hmm.. Yeah. Just read the compareTo documentation. We should be throwing NPE if either of them our null. Let me fix that and the documentation of the class, as recommended in Comparable interface